kubernetes/watcher.go

```diff
@@ -111,13 +116,28 @@ func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOption
 // client's workqueue that is used by the watcher. Workqueue name is important for exposing workqueue
 // metrics, if it is empty, its metrics will not be logged by the k8s client.
 func NewNamedWatcher(name string, client kubernetes.Interface, resource Resource, opts WatchOptions, indexers cache.Indexers) (Watcher, error) {
-	var store cache.Store
-	var queue workqueue.Interface
-	var cachedObject runtime.Object
 	informer, _, err := NewInformer(client, resource, opts, indexers)
 	if err != nil {
 		return nil, err
 	}
+	return NewNamedWatcherWithInformer(name, client, resource, informer, opts)
+}
+
+// NewNamedWatcherWithInformer initializes the watcher client to provide an events handler for
+// resource from the cluster (filtered to the given node) and also allows to name the k8s
+// client's workqueue that is used by the watcher. Workqueue name is important for exposing workqueue
+// metrics, if it is empty, its metrics will not be logged by the k8s client.
+// This function requires the underlying informer to be passed by the caller.
+func NewNamedWatcherWithInformer(
+	name string,
+	client kubernetes.Interface,
+	resource Resource,
+	informer cache.SharedInformer,
+	opts WatchOptions,
+) (Watcher, error) {
+	var store cache.Store
+	var queue workqueue.Interface
+	var cachedObject runtime.Object
 
 	store = informer.GetStore()
 	queue = workqueue.NewNamed(name)
```

Comment:
```suggestion
	queue = workqueue.NewWithConfig(QueueConfig{
		Name: name,
	})
```

nit: since you touch this code maybe you wanna deal with deprecated calls as well

Reply:
I actually want to avoid touching it as much as possible. I think the recommended way here is to used typed queues, and I don't want to make that kind of change in this PR.

Reply:
@swiatekm could you make an issue or PR (if it's quicker but I doubt that) to address the deprecated calls?  That way we don't lose track of this work.  Thanks.